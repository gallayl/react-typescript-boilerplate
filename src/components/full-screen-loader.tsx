import { CircularProgress } from '@material-ui/core'
import React from 'react'

export const FullScreenLoader: React.FunctionComponent = () => (
  <div
    style={{
      position: 'fixed',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    }}>
    <CircularProgress />
  </div>
)
