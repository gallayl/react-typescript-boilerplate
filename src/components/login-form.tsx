import React, { useState } from 'react'
import { Button, Divider, Paper, TextField, Typography } from '@material-ui/core'

export interface LoginFormProps {
  onLogin: (username: string, passowrd: string, repository: string) => void
}

export const LoginForm: React.FunctionComponent<LoginFormProps> = props => {
  const [userName, setUserName] = useState('')
  const [password, setPassword] = useState('')
  const [url, setUrl] = useState('')

  return (
    <div style={{ width: '100%', height: '100%', display: 'flex' }}>
      <Paper
        style={{ padding: '1em', flexShrink: 0, width: '450px', maxWidth: '90%', alignSelf: 'center', margin: 'auto' }}>
        <Typography variant="h4">Login</Typography>
        <form
          onSubmit={ev => {
            ev.preventDefault()
            props.onLogin(userName, password, url)
          }}>
          <Divider />
          <TextField
            required={true}
            margin="normal"
            label={'Username'}
            helperText={'The user login name'}
            fullWidth={true}
            value={userName}
            autoComplete="on"
            onChange={ev => {
              setUserName(ev.target.value)
            }}
          />
          <TextField
            required={true}
            margin="dense"
            label={'Password'}
            fullWidth={true}
            type="password"
            helperText={"The user's matching password to log in"}
            onChange={ev => {
              setPassword(ev.target.value)
            }}
          />
          <TextField
            margin="dense"
            required={true}
            label={'Repository URL'}
            helperText={'e.g. https://dev.demo.sensenet.com'}
            fullWidth={true}
            type="url"
            value={url}
            autoComplete="on"
            onChange={ev => {
              setUrl(ev.target.value)
            }}
          />
          <div style={{ display: 'flex', justifyContent: 'flex-end', marginTop: '1em' }}>
            <Button style={{ width: '100%' }} type="submit">
              <Typography variant="button">Log in</Typography>
            </Button>
          </div>
        </form>
      </Paper>
    </div>
  )
}
