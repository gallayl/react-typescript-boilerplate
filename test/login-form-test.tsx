import { shallow } from 'enzyme'
import React from 'react'
import { TextField } from '@material-ui/core'
import { LoginForm } from '../src/components/login-form'

describe('FullScreenLoader', () => {
  it('Matches snapshot', () => {
    const l = shallow(<LoginForm onLogin={() => {}} />)
    expect(l).toMatchSnapshot()
  })

  it('Matches snapshot with error', () => {
    const l = shallow(<LoginForm onLogin={() => {}} error=":(" />)
    expect(l).toMatchSnapshot()
  })

  it('executes onLogin after login', () => {
    const onLogin = jest.fn()
    const l = shallow(<LoginForm onLogin={onLogin} />)
    l.findWhere(e => e.type() === TextField).map(e =>
      e.prop('onChange')({
        preventDefault: () => undefined,
        target: { value: 'aaa' },
      }),
    )
    l.find('form').simulate('submit', { preventDefault: () => undefined })
    expect(onLogin).toBeCalled()
  })
})
