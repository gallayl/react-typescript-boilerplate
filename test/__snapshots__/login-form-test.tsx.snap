// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Matches snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <WithStyles(ForwardRef(Paper))
    style={
      Object {
        "alignSelf": "center",
        "flexShrink": 0,
        "margin": "auto",
        "maxWidth": "90%",
        "padding": "1em",
        "width": "450px",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
    <form
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(TextField))
        autoComplete="on"
        fullWidth={true}
        helperText="The user login name"
        label="Username"
        margin="normal"
        onChange={[Function]}
        required={true}
        value=""
      />
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        helperText="The user's matching password to log in"
        label="Password"
        margin="dense"
        onChange={[Function]}
        required={true}
        type="password"
      />
      <WithStyles(ForwardRef(TextField))
        autoComplete="on"
        fullWidth={true}
        helperText="e.g. https://dev.demo.sensenet.com"
        label="Repository URL"
        margin="dense"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "flex-end",
            "marginTop": "1em",
          }
        }
      >
        <WithStyles(ForwardRef(Button))
          style={
            Object {
              "width": "100%",
            }
          }
          type="submit"
        >
          <WithStyles(ForwardRef(Typography))
            variant="button"
          >
            Log in
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Button))>
      </div>
    </form>
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`Login Form Matches snapshot with error 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <WithStyles(ForwardRef(Paper))
    style={
      Object {
        "alignSelf": "center",
        "flexShrink": 0,
        "margin": "auto",
        "maxWidth": "90%",
        "padding": "1em",
        "width": "450px",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
    <form
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(TextField))
        autoComplete="on"
        fullWidth={true}
        helperText="The user login name"
        label="Username"
        margin="normal"
        onChange={[Function]}
        required={true}
        value=""
      />
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        helperText="The user's matching password to log in"
        label="Password"
        margin="dense"
        onChange={[Function]}
        required={true}
        type="password"
      />
      <WithStyles(ForwardRef(TextField))
        autoComplete="on"
        fullWidth={true}
        helperText="e.g. https://dev.demo.sensenet.com"
        label="Repository URL"
        margin="dense"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
      <WithStyles(ForwardRef(Typography))
        color="error"
      >
        :(
      </WithStyles(ForwardRef(Typography))>
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "flex-end",
            "marginTop": "1em",
          }
        }
      >
        <WithStyles(ForwardRef(Button))
          style={
            Object {
              "width": "100%",
            }
          }
          type="submit"
        >
          <WithStyles(ForwardRef(Typography))
            variant="button"
          >
            Log in
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Button))>
      </div>
    </form>
  </WithStyles(ForwardRef(Paper))>
</div>
`;
